modules/cnf-performing-end-to-end-tests-running-cyclictest.adoc

```diff
@@ -0,0 +1,144 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/cnf-performing-platform-verification-latency-tests.adoc
+
+:_content-type: PROCEDURE
+[id="cnf-performing-end-to-end-tests-running-cyclictest_{context}"]
+= Running cyclictest
+
+The `cyclictest` tool measures the real-time kernel scheduler latency on the specified CPUs.
+
+[IMPORTANT]
+====
+**Always** run the latency tests with `DISCOVERY_MODE=true` set. If you don't, the test suite will make changes to the running cluster configuration.
+====
+
+[NOTE]
+====
+When executing `podman` commands as a non-root or non-privileged user, mounting paths can fail with `permission denied` errors. To make the `podman` command work, append `:Z` to the volumes creation; for example, `-v $(pwd)/:/kubeconfig:Z`. This allows `podman` to do the proper SELinux relabeling.
+====
+
+.Prerequisites
+
+* You have logged in to `registry.redhat.io` with your Customer Portal credentials.
+
+* You have installed the real-time kernel in the cluster.
+
+* You have applied a cluster performance profile by using Node Tuning Operator.
+
+.Procedure
+
+* To perform the `cyclictest`, run the following command, substituting variable values as appropriate:
++
+[source,terminal,subs="attributes+"]
+----
+$ podman run -v $(pwd)/:/kubeconfig:Z -e KUBECONFIG=/kubeconfig/kubeconfig \
+-e LATENCY_TEST_RUN=true -e DISCOVERY_MODE=true -e ROLE_WORKER_CNF=worker-cnf \
+-e LATENCY_TEST_CPUS=10 -e LATENCY_TEST_RUNTIME=600 -e MAXIMUM_LATENCY=20 \
+registry.redhat.io/openshift4/cnf-tests-rhel8:v{product-version} \
+/usr/bin/test-run.sh -ginkgo.v -ginkgo.focus="cyclictest"
+----
++
+The command runs the `cyclictest` tool for 10 minutes (600 seconds). The test runs successfully when the maximum observed latency is lower than `MAXIMUM_LATENCY` (in this example, 20 μs). Latency spikes of 20 μs and above are generally not acceptable for telco RAN workloads.
++
+If the results exceed the latency threshold, the test fails.
++
+[IMPORTANT]
+====
+For valid results, the test should run for at least 12 hours.
+====
++
+.Example failure output
+[source,terminal,subs="attributes+"]
+----
+running /usr/bin/cnftests -ginkgo.v -ginkgo.focus=cyclictest
+I0908 13:01:59.193776      27 request.go:601] Waited for 1.046228824s due to client-side throttling, not priority and fairness, request: GET:https://api.compute-1.example.com:6443/apis/packages.operators.coreos.com/v1?timeout=32s
+Running Suite: CNF Features e2e integration tests
+=================================================
+Random Seed: 1662642118
+Will run 1 of 194 specs
+
+[...]
+
+Summarizing 1 Failure:
+
+[Fail] [performance] Latency Test with the cyclictest image [It] should succeed
+/remote-source/app/vendor/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/4_latency/latency.go:220
```

Comment:
here and in other places where we show the command's output as examples, the output is relevant for builds >4.10 (with node tuning operator)
since this applies to 4.10, let's preserve the 4.10 commands' output and change only the cnf-tests part impacted by the bug.

Reply:
Updated. PTAL now.

Reply:
@shajmakh Can you PTAL :)